### PR TITLE
OPS-5907: remove ParallelKey class

### PR DESCRIPTION
**Pull request by chaiken-verticloud**
This pull request removes the ParallelKey class as discussed in issue #956.  Performance tests on AWS with an m3.medium docker client and different size registry servers (m3.large, m3.xlarge, c3.xlarge) indicate that:
- the registry continues to work correctly
- the performance of the registry with ParallelKey deleted is no worse than the performance of the registry with ParallelKey

There may be some use cases where ParallelKey helps performance, but I couldn't find them with the test configuration using the m3.medium client.

### CODE REVIEW

File: depends/docker-registry-core/docker_registry/core/boto.py [hunk 1]
```diff
@@ -26,14 +26,8 @@
 
 """
 
-import gevent.monkey
-gevent.monkey.patch_all()
```
Comment (dmp42) [inline depends/docker-registry-core/docker_registry/core/boto.py]: I would not take the chance to remove gevent patching here - though it seems ludicrous, I'm pretty sure there was a reason for it being duplicated here.

**Comment by dmp42** [comment]
One simple nit, LGTM otherwise.

File: depends/docker-registry-core/docker_registry/core/boto.py (hunk 1 above)
Comment (chaiken-verticloud) [inline depends/docker-registry-core/docker_registry/core/boto.py]: Got it.  I'll put it back in.

**Comment by chaiken-verticloud** [comment]
I restored the gevent.monkey.patch code, which causes the following flake8 errors:
```
depends/docker-registry-core/docker_registry/core/boto.py:32:1: E402 module level import not at top of file
depends/docker-registry-core/docker_registry/core/boto.py:33:1: E402 module level import not at top of file
depends/docker-registry-core/docker_registry/core/boto.py:35:1: E402 module level import not at top of file
depends/docker-registry-core/docker_registry/core/boto.py:36:1: E402 module level import not at top of file
depends/docker-registry-core/docker_registry/core/boto.py:37:1: E402 module level import not at top of file
```
flake8 generated similar errors before I modified the file.

I'll test the code with the restored gevent.monkey.patch (even thought I don't expect any problems from it).

**Comment by chaiken-verticloud** [comment]
I tested the code with the restored gevent.monkey.patch with both [AWS_SECURE=false; STORAGE_REDIRECT=true] and [AWS_SECURE=true; STORAGE_REDIRECT=false].  Both worked as expected.  The two test runs seemed to indicate that there was no change in performance from the previously submitted version (but two runs are enough to verify performance).

**Comment by dmp42** [comment]
In order to make flake/hacking stop complaining here, you can:

- `gevent.monkey.patch_all()`
+`gevent.monkey.patch_all()  # noqa`

LGTM otherwise - merging.

**Comment by dmp42** [comment]
Fixed #956 and #954

**Pull request merged**